lib/ABI/FunctionType/Conversion.cpp

```diff
@@ -181,25 +182,78 @@ TCC::tryConvertingRegisterArguments(const ArgumentRegisters &Registers) {
 std::optional<llvm::SmallVector<model::Argument, 8>>
 TCC::tryConvertingStackArguments(model::QualifiedType StackArgumentTypes,
                                  size_t IndexOffset) {
+  // Qualifiers here are not allowed since this must point to a simple struct.
   revng_assert(StackArgumentTypes.Qualifiers().empty());
+
   auto *Unqualified = StackArgumentTypes.UnqualifiedType().get();
-  if (not Unqualified)
-    return {};
+  if (!Unqualified) {
+    // If there is no type, it means that the importer responsible for this
+    // function didn't detect any stack arguments and avoided creating
+    // a new empty type.
+    return llvm::SmallVector<model::Argument, 8>{};
+  }
+
+  // If the struct is empty, it indicates that there are no stack arguments.
+  const model::StructType &Types = *llvm::cast<model::StructType>(Unqualified);
+  if (Types.Fields().empty()) {
+    revng_assert(Types.size() == 0);
+    return llvm::SmallVector<model::Argument, 8>{};
+  }
 
-  auto *Pointer = llvm::dyn_cast<model::StructType>(Unqualified);
-  revng_assert(Pointer != nullptr,
-               "`RawFunctionType::StackArgumentsType` must be a struct");
-  const model::StructType &Types = *Pointer;
+  // Compute the alignment of the first argument.
+  auto CurrentAlignment = *ABI.alignment(Types.Fields().begin()->Type());
+  if (!isZeroOrPowerOfTwo(CurrentAlignment))
+    return std::nullopt;
```

Comment:
NOTE TO SELF: something seems off about this commit, I have probably lost a newer version of it when rebasing.

TODO: investigate.